### DISCUSSION TITLE
Is it possible to show 2 individual stacked bars in a merged form as a single bar as shown in the attached image ?

### DISCUSSION THREAD

**Discussion by vinaypuranik** [recharts/recharts#4762] [General]
![Screenshot 2024-07-12 at 19 04 13](https://github.com/user-attachments/assets/7a091275-790e-4df2-a8e0-5b3a2a753fa2)

**Comment by ckifer** [comment]
Hi! This would just be one stacked bar. What is the usecase for combining two stacks into one if they aren't one stack?

**Comment by vinaypuranik** [comment]
If you see the figure, yellow isn't stacked. It begins from zero, same like green.  Its like a bar with two unstacked values and two stacked values, if you see the merged form. 

From the documentation its not clear on how to control the stack order plus what if  I want 1 value to stack and other to be unstacked meaning it should start from zero not on top of another .

**Reply by ckifer** [reply]
Sorry I guess I didn't understand your original question. Are you saying that "UI Result" is what you want, or the result you got from trying?

**Reply by ckifer** [reply]
https://codesandbox.io/p/sandbox/bar-chart-has-background-forked-qwqmyq

is this what you want? If not can you fork that sandbox to represent what you are trying to do

stack order is controlled by render order

If you want something to start at 0 don't add a stackId

**Reply by vinaypuranik** [reply]
In a single bar, I want to show 4 values, in that 2 should start from 0 and 2 should be stacked, if I dont give stack Id it will render different bar

**Reply by ckifer** [reply]
Could you explain why you need it like this? I can't see how any data would make sense to be represented this way. 

In a stack everything is always relative to the previous value. 

Otherwise, wouldn't the user be confused that the bar shows on the YAxis from a certain value to a certain but the value actually starts at 0?

Do you have an example where this works in a different charting library or BI solution? I'd imagine it isn't possible there either

**Reply by vinaypuranik** [reply]
Could you explain why you need it like this? I can't see how any data would make sense to be represented this way.
>> this is requirement from business/ UI/UX designer. I dont have much say in this. 

So far, I managed to construct the data in such a way that it stack order is dynamically rendered based on the data.
![Screenshot 2024-07-15 at 15 00 21](https://github.com/user-attachments/assets/e606f038-2618-4c26-92c2-c1a07d73d7bd)


 And as for starting values from 0, have a separate calculations for it.. for example, from the first figure attached in description if the output (right hand side of the figure) was stacked bar then yellow value would be actualYellow - green-purple.  (left side from the figure)


But, this messed up the legend colors. Need to investigate why its rendered as black.
